pca_ica/run_pca.m

```diff
@@ -54,6 +63,22 @@ function run_pca(movie_source, num_PCs, varargin)
     fprintf('%s: Finished median filtering!\n', datestr(now));
 end
 
+% Calculate max-projection (required for trimming and auto-setting #of PCs)
+if do_trim || autoset_num_PCs
+    max_proj = max(M,[],3);
+end
+
+% Detect local maxima in max-projection image to get an estimated maximum 
+%number of cells in the movie
+if autoset_num_PCs
+    cents = local_maxima_2D(max_proj);
+    num_PCs = size(cents,2);
+    if num_PCs>max_num_PCs

```

Comment:
In such cases, it would be helpful to write to console that the automatically determined number of PCs has been overwritten, since it may indicate that the movie is somewhat weird.